File: Rest-Api-V3/orders-private/orders-place/orders-place.js
const axios = require('axios');
const crypto = require('crypto');
require('dotenv').config();

async function placeOrders(orders) {
    const apiKey = process.env.API_KEY;
    const secretKey = process.env.API_SECRET;
    const ts = new Date().toISOString();
    const nonce = Math.random().toString(36).substring(2);
    const method = "/v3/orders/place";
    const apiUrl = "api.ox.fun";

    const postData = {
        recvWindow: 20000,
        responseType: "FULL",
        timestamp: Date.now(),
        orders: orders
    };

    const body = JSON.stringify(postData);
    const msgString = `${ts}\n${nonce}\nPOST\n${apiUrl}\n${method}\n${body}`;

    const sign = crypto.createHmac('sha256', secretKey)
        .update(msgString)
        .digest('base64');

    const headers = {
        'Content-Type': 'application/json',
        'AccessKey': apiKey,
        'Timestamp': ts,
        'Signature': sign,
        'Nonce': nonce
    };

    try {
        const response = await axios.post(`https://${apiUrl}${method}`, body, { headers });
        console.log('Response:', response.data);
    } catch (error) {
        console.error('Error placing orders:', error.response ? error.response.data : error.message);
    }
}

// Example usage
const orders = [
    {
        clientOrderId: 1612249737724,
        marketCode: "BTC-USD-SWAP-LIN",
        side: "SELL",
        quantity: "0.001",
        timeInForce: "GTC",
        orderType: "LIMIT",
        price: "50007"
    },
    {
        clientOrderId: 1612249737724,
        marketCode: "BTC-USD-SWAP-LIN",
        side: "BUY",
        quantity: "0.002",
        timeInForce: "GTC",
        orderType: "LIMIT",
        price: "54900"
    }
];

placeOrders(orders);

File: Rest-Api-V3/orders-private/orders-place/orders-place.py
import os
import requests
import hmac
import hashlib
import base64
import json
import time
from dotenv import load_dotenv

load_dotenv()

def place_orders(orders):
    api_key = os.getenv('API_KEY')
    secret_key = os.getenv('API_SECRET').encode('utf-8')
    ts = time.strftime('%Y-%m-%dT%H:%M:%S', time.gmtime())
    nonce = str(int(time.time() * 1000))
    method = "/v3/orders/place"
    api_url = "api.ox.fun"

    post_data = {
        "recvWindow": 20000,
        "responseType": "FULL",
        "timestamp": int(time.time() * 1000),
        "orders": orders
    }

    body = json.dumps(post_data)
    msg_string = f"{ts}\n{nonce}\nPOST\n{api_url}\n{method}\n{body}"

    sign = base64.b64encode(hmac.new(secret_key, msg_string.encode('utf-8'), hashlib.sha256).digest()).decode('utf-8')

    headers = {
        'Content-Type': 'application/json',
        'AccessKey': api_key,
        'Timestamp': ts,
        'Signature': sign,
        'Nonce': nonce
    }

    try:
        response = requests.post(f"https://{api_url}{method}", data=body, headers=headers)
        response.raise_for_status()
        print('Response:', response.json())
    except requests.exceptions.RequestException as error:
        print('Error placing orders:', error)

# Example usage
orders = [
    {
        "clientOrderId": 1612249737724,
        "marketCode": "BTC-USD-SWAP-LIN",
        "side": "SELL",
        "quantity": "0.001",
        "timeInForce": "GTC",
        "orderType": "LIMIT",
        "price": "50007"
    },
    {
        "clientOrderId": 1612249737724,
        "marketCode": "BTC-USD-SWAP-LIN",
        "side": "BUY",
        "quantity": "0.002",
        "timeInForce": "GTC",
        "orderType": "LIMIT",
        "price": "54900"
    }
]

place_orders(orders)